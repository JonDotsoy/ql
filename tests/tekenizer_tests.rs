#[macro_use]
extern crate assert_matches;

#[cfg(test)]
mod tekenizer_tests {

    use insta::assert_debug_snapshot;
    use ql::QL;

    #[test]
    fn tekenizer() {
        let payload = r#"   SELECT  * FROM cars   "#;

        let tokens = QL::lexer(payload);

        assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_2() {
        let payload = r#"SELECT"#;

        let tokens = QL::lexer(payload);

        assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_string() {
        let payload = r#" "any world" try"#;

        let ref tokens = QL::lexer(payload);

        assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_numeric() {
        let payload = r#"9"#;

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_numeric_2() {
        let payload = r#"9.12"#;

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_numeric_3() {
        let payload = r#"9_123_123.122_123_943"#;

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_parenthesis_and_bracket() {
        let payload = r#"(var_1)[var_2]"#;

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_newline() {
        let payload = "abc\ndef";

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_newline_2() {
        let payload = "abc\r\ndef";

        let ref tokens = QL::lexer(payload);

        // assert_matches!(tokens, Ok(_));
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tekenizer_lexer_query_1() {
        let payload = "ctx.var_name:123 ";
        let ref tokens = QL::lexer(payload);
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tokenizer_lexer_sample_1() {
        let payload = "search value";
        let ref tokens = QL::lexer(payload);
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tokenizer_lexer_sample_2() {
        let payload = "search value condition:";
        let ref tokens = QL::lexer(payload);
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tokenizer_lexer_sample_3() {
        let payload = r#"search value condition: "value""#;
        let ref tokens = QL::lexer(payload);
        assert_debug_snapshot!(tokens);
    }

    #[test]
    fn tokenizer_lexer_sample_4() {
        let payload = r#"
            search value
            condition: "value"
            context.condition > "value"
        "#;
        let ref tokens = QL::lexer(payload);
        assert_debug_snapshot!(tokens);
    }
}
