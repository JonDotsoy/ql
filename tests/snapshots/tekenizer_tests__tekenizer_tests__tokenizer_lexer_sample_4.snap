---
source: tests/tekenizer_tests.rs
expression: tokens
---
Ok(
    [
        Token {
            kind: "newline",
            raw: "\n",
            span: Span {
                start: 0,
                end: 1,
            },
        },
        Token {
            kind: "keyword",
            raw: "search",
            span: Span {
                start: 13,
                end: 19,
            },
        },
        Token {
            kind: "keyword",
            raw: "value",
            span: Span {
                start: 20,
                end: 25,
            },
        },
        Token {
            kind: "newline",
            raw: "\n",
            span: Span {
                start: 25,
                end: 26,
            },
        },
        Token {
            kind: "keyword",
            raw: "condition",
            span: Span {
                start: 38,
                end: 47,
            },
        },
        Token {
            kind: "colon",
            raw: ":",
            span: Span {
                start: 47,
                end: 48,
            },
        },
        Token {
            kind: "string",
            raw: "value",
            span: Span {
                start: 50,
                end: 55,
            },
        },
        Token {
            kind: "newline",
            raw: "\n",
            span: Span {
                start: 56,
                end: 57,
            },
        },
        Token {
            kind: "keyword",
            raw: "context",
            span: Span {
                start: 69,
                end: 76,
            },
        },
        Token {
            kind: "dot",
            raw: ".",
            span: Span {
                start: 76,
                end: 77,
            },
        },
        Token {
            kind: "keyword",
            raw: "condition",
            span: Span {
                start: 77,
                end: 86,
            },
        },
        Token {
            kind: "operation",
            raw: ">",
            span: Span {
                start: 87,
                end: 88,
            },
        },
        Token {
            kind: "string",
            raw: "value",
            span: Span {
                start: 90,
                end: 95,
            },
        },
        Token {
            kind: "newline",
            raw: "\n",
            span: Span {
                start: 96,
                end: 97,
            },
        },
    ],
)
